### PR TITLE
FALCON-2223 Distributed mode support for User Extension

### PR DESCRIPTION
**Pull request by PraveenAdlakha**

### CODE REVIEW

File: webapp/src/main/java/org/apache/falcon/resource/ExtensionManager.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ */
+@Path("extension")
+public class ExtensionManager {
+    public static final Logger LOG = LoggerFactory.getLogger(ExtensionManager.class);
+
+    @GET
+    @Path("enumerate")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getExtensions() {
+        LOG.error("Enumerate is not supported on Server.Please run your operation on prism ");
```
Comment (pallavi-rao) [inline webapp/src/main/java/org/apache/falcon/resource/ExtensionManager.java]: Nit : For this, and all the messages below fix typos and spaces : "Enumerate is not supported on Server. Please run your operation on Prism."

File: src/conf/startup.properties [hunk 2]
```diff
@@ -56,7 +56,7 @@
 #org.apache.falcon.metadata.MetadataMappingService,\
 
 ##Add if you want to use Trusted or User Extensions
-## In case of distributed Mode enable ExtensionService only on Prism
+## In case of distributed Mode enable ExtensionService only on Prism in prism.application.services
```
Comment (sandeepSamudrala) [inline src/conf/startup.properties]: via prism.application.services

**Comment by pallavi-rao** [comment]
@PraveenAdlakha, Please resolve the merge conflicts

File: prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+    public static void validateExtensionName(final String extensionName) {
+        if (StringUtils.isBlank(extensionName)) {
+            throw FalconWebException.newAPIException("Extension name is mandatory and shouldn't be blank",
+                    Response.Status.BAD_REQUEST);
+        }
+    }
+
+    public String registerExtensionMetadata(String extensionName, String path, String description) {
+        validateExtensionName(extensionName);
```
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java]: validation is happening at the prism anyways. so not required to check again over here.

Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java]: Same for other apis too. Either remove it from the prxoy server api or from the super class' api

**Comment by PraveenAdlakha** [comment]
@pallavi-rao resolved.

**Comment by sandeepSamudrala** [comment]
Looks Good. :+1:

**Comment by PraveenAdlakha** [comment]
all checks have completed successfully will be merging it.

**Pull request closed without merging**